### DISCUSSION TITLE
Build or-tools (Vehicle Routing Problem) for windows with .Net Framework 4.6.2

### DISCUSSION THREAD

**Discussion by vaibhav21791** [google/or-tools#4190] [Build and configuration issues]
I want to build Google Or-tools to use it for solving vehicle routing problem. I'm not able to build it on the Windows machine. I have installed necessary prerequisites but still it is failing. The error says "Could NOT find ZLIB(missing: ZLIB_LIBRARY ZLIB_INCLUDE_DTR)".  My end goal is to take an open source project for Windows (.NET) and build it to make it compatible with .Net Framework 4.6.2. Current stable version is compatible with .NET 6. Any help would be appreciated.

**Comment by lperron** [comment]
did you use cmake ?
have you added the -DBUILD_DOTNET=ON ?

**Comment by lperron** [comment]
Why do you need to build ?

.NEt 462 is included in or-tools 9.9 .

**Comment by lperron** [comment]
https://nuget.info/packages/Google.OrTools/9.9.3963

**Reply by vaibhav21791** [reply]
I am able to build the project using (Googl.OrTools/9.9.3963 of version .NETFramework, version=v4.6.2)  and its dependent assembly https://nuget.info/packages/Google.Protobuf/3.25.3 of version .NETFramework, version=v4.5. However, it is giving me the following error, "The type initializer for 'Google.OrTools.ConstraintSolver.operations_research_constraint_solverPINVOKE' threw an exception"
DllNotFoundException: Unable to load DLL 'google-ortools-native': The specified module could not be found. (Exception from HRESULT: 0x8007007E).

I'm using .NET Framework 4.6.2. Could you please help me on this?

**Comment by Mizux** [comment]
If you really want to build it on windows only for net462 take a look here: aka `-DUSE_DOTNET_462=ON` to add to the cmake configure command line...
https://github.com/google/or-tools/blob/5124f4975bd6ab5a863ea55657833ab640730429/CMakeLists.txt#L322-L323

But as laurent say nuget package should support it but since we use .Net Core CLI so you may have to manually copy some .dll in the target dir if I recall correctly since net462 was prior this ecosystem and miss some csproj/msbuild properties...
